services/promtail.nix

```diff
@@ -0,0 +1,99 @@
+{ config, lib, pkgs, ... }:
+
+let
+  prettyJSON = conf:
+    pkgs.runCommand "promtail-config.json" { } ''
+      echo '${builtins.toJSON conf}' | ${pkgs.jq}/bin/jq 'del(._module)' > $out
+    '';
+  dataDir = "/var/lib/promtail";
+  configuration = {
+    server = {
+      http_listen_port = 9080;
+    };
+    clients = [{
+      url = "http://hardcase.internal:3100/loki/api/v1/push";
+    }];
+    scrape_configs = [
+      {
+        job_name = "syslog";
+        syslog = {
+          listen_address = "0.0.0.0:514";
+          idle_timeout = "60s";
+          label_structured_data = true;
+          labels = {
+            job = "syslog";
+          };
+        };
+        relabel_configs = [
+          {
+            source_labels = [ "__syslog_message_hostname" ];
+            target_label = "host";
+          }
+          {
+            source_labels = [ "__syslog_connection_ip_address" ];
+            target_label = "ip";
+          }
+          {
+            source_labels = ["__syslog_message_app_name"];
+            target_label = "app";
+          }
+          {
+            source_labels = ["__syslog_message_severity"];
+            target_label = "level";
+          }
+        ];
+      }
+      # {
```

Comment:
Do we want to keep this block of comments?

Reply:
I'm not sure what to do with the block, I wonder if it might be useful at some point but aside from putting it in an unused variable, I'm unsure where to put it

Reply:
This will be useful, I think it's fine to keep in